httpx/_client.py

```diff
@@ -760,15 +760,12 @@ def _send_handling_auth(
         auth: Auth,
         timeout: Timeout,
     ) -> Response:
-        if auth.requires_request_body:
-            request.read()
+        auth_flow = auth.sync_auth_flow(request)
+        request = auth_flow.send(None)  # type: ignore
```

Comment:
Wondering which of `auth_flow.send(None)` vs `next(auth_flow)` we should prefer for this? I don't really mind either way, but I guess whichever we stick with we should use consistently across:

* The docs.
* The client implementation.
* The auth implementation.

Reply:
I'll stick to our current usage of `next()` / `.__anext__()` for the purpose of this PR. Given that this works more nicely with type checking it's a pretty okay approach.